### DISCUSSION TITLE
How to prevent persistent data from being modified by users

### DISCUSSION THREAD

**Discussion by Ryongyon** [pmndrs/zustand#1322] [Q&A]
Should i use `getStorage` or `merge`? Or I didn't think of a better implementation plan?

**Comment by dai-shi** [comment]
Does being modified by users mean mutating localStorage manually or another code or something?

**Reply by Ryongyon** [reply]
What I want to do is if the user manually changes the localStorage, when merging persistent data, I can check whether the data conforms to the expected value, and if not, use the initial value

**Reply by dai-shi** [reply]
Sounds like `migrate` would help, but not sure. @AnatoleLucet thoughts?

**Reply by Ryongyon** [reply]
I tried `migrate`, but I found that I must use it together with `version `to expect work

**Reply by dai-shi** [reply]
Ah, I see. Then, `deserialize` might be a better option.
